modules/nw-udn-cr.adoc

```diff
@@ -34,32 +34,89 @@ spec:
+* `Layer2` subnets may be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
+
++
+[NOTE]
+====
+There are two additional configurations that UDN uses, `spec.joinSubnets` and `spec.IPAMlifecycle`. It is not recommended to set these fields without explicit need and understanding of the OVN network topology.
+
+When omitted, the platform will set default values for the `joinSubnets` field of `100.65.0.0/16` for IPv4 and  `fd99::/64` for IPv6. If the default address values are used anywhere in the cluster's network you must override it by setting the `joinSubnets` field. If you choose to set this field, ensure it does not conflict with other subnets in the cluster such as the cluster subnet, the `default` network cluster subnet, and the masquerade subnet.
+
+The `joinSubnets` field configures the routing between different segments within the Layer 2 network. Dual-stack clusters may set 2 subnets, one for each IP family, otherwise only 1 subnet is allowed. This field is only allowed for the `Primary` network.
+
+The `IPAMLifecycle` field configures the IP address management system (IPAM). You might use this field for virtual workloads to ensure persistent IP addresses. This field is allowed when `topology` is `layer2`. The `subnets` field must be specified when this field is specified. Setting a value of `Persistent` ensures that your virtual workloads have persistent IP addresses. These are assigned by the container network interface (CNI) and used by OVN-Kubernetes to program pod IP addresses. You must not change this for pod annotations.
+====
+
+.. Create a YAML file, such as `my-layer-three-udn.yaml`, to define your request as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-2-primary # <1>
+  namespace: <some_custom_namespace>
+spec:
+  topology: Layer3 # <2>
+  layer3: # <3>
+    role: Primary # <4>
+    mtu: 1300 # <5>
+    subnets: # <6>
+      - cidr: 10.150.0.0/16
+        hostSubnet: 24
+      - cidr: 2001:db8::/60
+        hostSubnet: 64
+----
+<1> Name of your `UserDefinedNetwork` resource. This should not be `default` or duplicate any global namespaces created by the Cluster Network Operator (CNO).
+<2> The `topology` field describes the network configuration, accepted values are `Layer2` and `Layer3`. Specifying a `Layer3` topology type creates a layer 2 segment per node, each with a different subnet. Layer 3 routing is used to interconnect node subnets.
+<3> This field specifies the topology configuration, valid values are `layer2` or `layer3`.
+<4> Specifies a `Primary` or `Secondary` role. `Primary` is the only `role` specification supported in {product-version}.
```

Comment:
Be careful with {product-version} here. This will carry over into 4.18 and say "4.18" in that version. not sure if that's what you intend.